csharp/src/SeedLang/Ast/Executor.cs

```diff
@@ -39,6 +43,26 @@ internal sealed class Executor : AstWalker {
       Visit(node);
     }
 
+    // Calls a AST function with given arguments.
+    internal Value Call(FunctionStatement func, Value[] arguments) {
+      if (func.Parameters.Length != arguments.Length) {
+        throw new DiagnosticException(SystemReporters.SeedAst, Severity.Fatal, "", null,
+                                      Message.RuntimeErrorIncorrectArgsCount);
+      }
+      _env.EnterScope();
+      for (int i = 0; i < func.Parameters.Length; i++) {
+        _env.SetVariable(func.Parameters[i], arguments[i]);
```

Comment:
Here is a question not quite related to this pr: have you considered the future improvement to support closures and coroutines? For example, Lua has an "upvalue" design to support closures and co-routines...

I think we just need to make sure the current design is extendable to be upgraded to a closure-friendly design in the future.

Reply:
I have plan to implement "upvalue". It's not too hard, just need add a Upvalue value type, and store the outside stack values into it in the returned closure. We can implement it in the future.

Reply:
Great!